LibAddonMenu-2.0/LibAddonMenu-2.0/LibAddonMenu-2.0.lua

```diff
@@ -1,1317 +1,1387 @@
+    logger = LibDebugLogger(MAJOR)
+end
+
+if LAMSettingsPanelCreated and not LAMCompatibilityWarning then
+    PrintLater("An old version of LibAddonMenu with compatibility issues was detected. For more information on how to proceed search for LibAddonMenu on esoui.com")
+    LAMCompatibilityWarning = true
+end
+
+--UPVALUES--
+local wm = WINDOW_MANAGER
+local em = EVENT_MANAGER
+local sm = SCENE_MANAGER
+local cm = CALLBACK_MANAGER
+local tconcat = table.concat
+local tinsert = table.insert
+
+local MIN_HEIGHT = 26
+local HALF_WIDTH_LINE_SPACING = 2
+local OPTIONS_CREATION_RUNNING = 1
+local OPTIONS_CREATED = 2
+local LAM_CONFIRM_DIALOG = "LAM_CONFIRM_DIALOG"
+local LAM_DEFAULTS_DIALOG = "LAM_DEFAULTS"
+local LAM_RELOAD_DIALOG = "LAM_RELOAD_DIALOG"
+
+local addonsForList = {}
+local addonToOptionsMap = {}
+local optionsState = {}
+lam.widgets = lam.widgets or {}
+local widgets = lam.widgets
+lam.util = lam.util or {}
+local util = lam.util
+lam.controlsForReload = lam.controlsForReload or {}
+local controlsForReload = lam.controlsForReload
+
+local function GetDefaultValue(default)
+    if type(default) == "function" then
+        return default()
+    end
+    return default
+end
+
+local function GetStringFromValue(value)
+    if type(value) == "function" then
+        return value()
+    elseif type(value) == "number" then
+        return GetString(value)
+    end
+    return value
+end
+
+local FAQ_ICON_COLOR = ZO_ColorDef:New("FFFFFF") -- white
+local FAQ_ICON_MOUSE_OVER_COLOR = ZO_ColorDef:New("B8B8D3") -- dark-blue/white
+local HELP_URL_TOOLTIP
+
+local function GetColorForState(disabled)
+    return disabled and ZO_DEFAULT_DISABLED_COLOR or ZO_DEFAULT_ENABLED_COLOR
+end
+
+local function CreateFAQTexture(parent, controlData, controlName, anchorPoint, anchorRelativeControl, anchorRelativePoint, anchorX, anchorY)
+    anchorPoint = anchorPoint or TOPLEFT
+    anchorRelativeControl = anchorRelativeControl or parent
+    anchorRelativePoint = anchorRelativePoint or TOPLEFT
+    anchorX = anchorX or 0
+    anchorY = anchorY or 0
+    local onMouseExitAlpha = 0.4
+
+    local helpUrl = controlData and controlData.helpUrl and GetStringFromValue(controlData.helpUrl)
+    if not helpUrl or helpUrl == "" then return end
+    local faqControl = wm:CreateControl(nil, parent, CT_TEXTURE)
+    faqControl:SetHidden(false)
+    faqControl:ClearAnchors()
+    faqControl:SetDrawLayer(DL_OVERLAY)
+    faqControl:SetTexture("\\ESOUI\\ART\\miscellaneous\\help_icon.dds") --small (i): \\ESOUI\\ART\\help\\help_tabicon_tutorial_up.dds
+    faqControl:SetDimensions(23, 23)
+
+    faqControl:SetColor(FAQ_ICON_COLOR:UnpackRGBA())
+    faqControl:SetAlpha(onMouseExitAlpha)
+    faqControl:SetAnchor(anchorPoint, anchorRelativeControl, anchorRelativePoint, anchorX, anchorY-1)
+
+    faqControl.isMouseOver = false
+    faqControl.data = faqControl.data or {}
+    faqControl.data.helpUrl = helpUrl
+    --faqControl.data.tooltipText = HELP_URL_TOOLTIP:format(helpUrl)

```

Comment:
Not sure if this is supposed to be commented out?
I also don't like how HELP_URL_TOOLTIP spreads this out into 3 places. Maybe just write `("%s: %s"):format(util.L.WEBSITE, helpUrl)` instead?

Reply:
I thought having 1 variable would be less often call the string.format stuff. Then I realized that the localization is done further down in the files and not at the beginning, so the access to util.L and L wa snot given at the top.
So whatever you want... :-)